src/notebook.c

```diff
@@ -455,6 +455,34 @@ static void on_close_documents_right_activate(GtkMenuItem *menuitem, GeanyDocume
 }
 
 
+static void on_copy_basename_to_clipboard_activate(GtkMenuItem *menuitem, GeanyDocument *doc)
+{
+	g_return_if_fail(doc->is_valid);
+	
+	gchar *bname = g_path_get_basename(doc->real_path);
+	gtk_clipboard_set_text(gtk_clipboard_get(GDK_NONE), bname, -1);
```

Comment:
I could be mistaken, but I think there might be a subtle encoding bug here (and the other two new functions).

`doc->real_path` is in "locale" encoding and `g_path_get_basename()` will spit out "locale" encoding, but `gtk_clipboard_set_text()` is expecting the 2nd argument to be in UTF-8 encoding. If I'm not mistaken, it might need to be something like this (untested):

```c
gchar *bname = g_path_get_basename(doc->real_path);
gchar *utf8_bname = utils_get_utf8_from_locale(bname);
gtk_clipboard_set_text(gtk_clipboard_get(GDK_NONE), utf8_bname, -1);
g_free(utf8_bname);
g_free(bname);
```

Reply:
I take @elextr's +1/thumbs-up that I may be correct.

I think it should be like this:
```diff
diff --git a/src/notebook.c b/src/notebook.c
index cef5375f9..fb47ec29d 100644
--- a/src/notebook.c
+++ b/src/notebook.c
@@ -460,8 +460,10 @@ static void on_copy_basename_to_clipboard_activate(GtkMenuItem *menuitem, GeanyD
 	g_return_if_fail(doc->is_valid);
 	
 	gchar *bname = g_path_get_basename(doc->real_path);
-	gtk_clipboard_set_text(gtk_clipboard_get(GDK_NONE), bname, -1);
+	gchar *utf8_bname = utils_get_utf8_from_locale(bname);
 	g_free(bname);
+	gtk_clipboard_set_text(gtk_clipboard_get(GDK_NONE), utf8_bname, -1);
+	g_free(utf8_bname);
 }
 
 
@@ -469,7 +471,9 @@ static void on_copy_full_path_to_clipboard_activate(GtkMenuItem *menuitem, Geany
 {
 	g_return_if_fail(doc->is_valid);
 	
-	gtk_clipboard_set_text(gtk_clipboard_get(GDK_NONE), doc->real_path, -1);
+	gchar *utf8_fname = utils_get_utf8_from_locale(doc->real_path);
+	gtk_clipboard_set_text(gtk_clipboard_get(GDK_NONE), utf8_fname, -1);
+	g_free(utf8_fname);
 }
 
 
@@ -478,8 +482,10 @@ static void on_copy_dir_path_to_clipboard_activate(GtkMenuItem *menuitem, GeanyD
 	g_return_if_fail(doc->is_valid);
 
 	gchar *dirpath = g_path_get_dirname(doc->real_path);
-	gtk_clipboard_set_text(gtk_clipboard_get(GDK_NONE), dirpath, -1);
+	gchar *utf8_dirpath = utils_get_utf8_from_locale(dirpath);
 	g_free(dirpath);
+	gtk_clipboard_set_text(gtk_clipboard_get(GDK_NONE), utf8_dirpath, -1);
+	g_free(utf8_dirpath);
 }
 
 
```